src/main/java/com/thirdcc/webapp/service/impl/EventAttendeeServiceImpl.java

```diff
@@ -42,9 +56,36 @@ public EventAttendeeServiceImpl(EventAttendeeRepository eventAttendeeRepository,
     @Override
     public EventAttendeeDTO save(EventAttendeeDTO eventAttendeeDTO) {
         log.debug("Request to save EventAttendee : {}", eventAttendeeDTO);
-        EventAttendee eventAttendee = eventAttendeeMapper.toEntity(eventAttendeeDTO);
-        eventAttendee = eventAttendeeRepository.save(eventAttendee);
-        return eventAttendeeMapper.toDto(eventAttendee);
+
+        Set<EventStatus> eventStatuses = new HashSet<EventStatus>() {{
+            add(EventStatus.OPEN);
+            add(EventStatus.POSTPONED);
+        }};
+
+        userRepository
+            .findOneById(eventAttendeeDTO.getUserId())
+            .orElseThrow(() -> new BadRequestException("User not found"));
+
+        Event event = eventRepository
+            .findOneByIdAndStatusIn(eventAttendeeDTO.getEventId(), eventStatuses)
+            .orElseThrow(()-> new BadRequestException("This event does not exists or it is not happening"));
+
+        if(event.getEndDate().isBefore(Instant.now())){
+            throw new BadRequestException("Cannot add attendee to ended event");
+        }
+
+        boolean eventAttendeeExisted = eventAttendeeRepository
+            .findOneByEventIdAndUserId(eventAttendeeDTO.getEventId(),eventAttendeeDTO.getUserId())
+            .isPresent();
+        log.debug("eventAttendeeExisted:"+ eventAttendeeExisted);
+
+        if(eventAttendeeExisted){
+            throw new BadRequestException("User has registered as attendee for this event");
+        } else {
```

Comment:
`else` block is actually not required.

Reply:
Noted, the code within the else block is moved out, and the else block is deleted.